components/cam/src/physics/crm/sam/crm_module.F90

```diff
@@ -230,21 +301,21 @@ subroutine crm(lchnk, ncrms, dt_gl, plev,       &
   call allocate_scalar_momentum(ncrms)
 #endif
 
-  crm_rad_temperature => crm_rad%temperature(1:ncrms,:,:,:)
-  crm_rad_qv          => crm_rad%qv         (1:ncrms,:,:,:)
-  crm_rad_qc          => crm_rad%qc         (1:ncrms,:,:,:)
-  crm_rad_qi          => crm_rad%qi         (1:ncrms,:,:,:)
-  crm_rad_cld         => crm_rad%cld        (1:ncrms,:,:,:)
-  crm_rad_qrad        => crm_rad%qrad       (1:ncrms,:,:,:)
-
-  crm_state_u_wind      => crm_state%u_wind     (1:ncrms,:,:,:)
-  crm_state_v_wind      => crm_state%v_wind     (1:ncrms,:,:,:)
-  crm_state_w_wind      => crm_state%w_wind     (1:ncrms,:,:,:)
-  crm_state_temperature => crm_state%temperature(1:ncrms,:,:,:)
-  crm_state_qt          => crm_state%qt         (1:ncrms,:,:,:)
-  crm_state_qp          => crm_state%qp         (1:ncrms,:,:,:)
-  crm_state_qn          => crm_state%qn         (1:ncrms,:,:,:)
-  
+  crm_rad_temperature = crm_rad%temperature(1:ncrms,1:crm_nx_rad,1:crm_ny_rad,1:crm_nz)
```

Comment:
I think I like this approach. I chose to go this route with the C++ code. This will require extra copies, but it shields the OpenACC and OpenMP runtimes from the horrors of pbuf, which is a good thing.

Since you're doing this, however, I think it woudl be best to leave all of the data structures alone, meaning crm_output, crm_input, crm _rad, crm_state, etc. Since you're already creating a contiguous data structure to copy the data into, you don't need to get rid of the classes and pointers being passed into the `crm(...)` call. @whannah1  and @brhillman , what do y'all think?